libs/product/src/reducers/configurable-product-entities/configurable-product-entities-reducer-adapter.ts

```diff
@@ -0,0 +1,12 @@
+import { EntityAdapter, createEntityAdapter } from '@ngrx/entity';
+
+import { DaffProductVariantAttributesDictionary } from '../../models/configurable-product';
+
+/**
+ * Configurable Product Applied Attributes Adapter for changing/overwriting entity state.
+ */
+export const daffConfigurableProductAppliedAttributesEntitiesAdapter = (() => {
+	let cache;
+  return (): EntityAdapter<DaffProductVariantAttributesDictionary> =>
+    cache = cache || createEntityAdapter<DaffProductVariantAttributesDictionary>();
```

Comment:
Does this need to be generic?

Reply:
I don't think so. These "entities" aren't really entities. They're really just a way to keep track of a bunch of on/off flags for each configurable product, and it wouldn't make sense to add to the model.

Reply:
Well the attribute values have significance beyond on/off, right? Like one of these entites might look like
```ts
{
  color: 'blue',
  size: 'M'
}
```

I agree that it wouldn't really be that useful to customize the model though so a generic would not be appropriate here.

Reply:
yea, that's right